src/crypto/abe/hybrid_crypto/cleartext_header.ts

```diff
@@ -0,0 +1,85 @@
+import { logger } from "../../../utils/logger"
+import { Metadata } from "./metadata"
+import { deserializeList, fromBeBytes } from "../../../utils/utils"
+import { SYMMETRIC_KEY_SIZE } from "./encryption_parameters"
+
+export class ClearTextHeader {
+  private _symmetricKey: Uint8Array
+  private _metadata: Metadata
+
+  /* Getter / Setters */
+  public get symmetricKey(): Uint8Array {
+    return this._symmetricKey
+  }
+  public set symmetricKey(value: Uint8Array) {
+    this._symmetricKey = value
+  }
+  public get metadata(): Metadata {
+    return this._metadata
+  }
+  public set metadata(value: Metadata) {
+    this._metadata = value
+  }
+
+  constructor(symmetricKey: Uint8Array, metadata: Metadata) {
+    this._symmetricKey = symmetricKey
+    this._metadata = metadata
+  }
+
+  /**
+   * Deserialize a clear text header.
+   *
+   * @param cleartextHeader a raw header
+   * @returns the object deserialized
+   */
+  public static parseJson(cleartextHeader: Uint8Array): ClearTextHeader {
+    const clearTextHeaderString = new TextDecoder().decode(cleartextHeader.buffer)
+    logger.log(() => "clearTextHeaderString: " + clearTextHeaderString)
+    // parse JSON to object
+    const clearTextHeaderJson = JSON.parse(clearTextHeaderString)
+    logger.log(() => "clearTextHeader: " + clearTextHeaderJson)
+    logger.log(() => "symmetric_key: " + clearTextHeaderJson.symmetric_key)
```

Comment:
Isn't it dangerous to log a symmetric key?

Reply:
done